src/cirrus/builtins/functions/process/lambda_function.py

```diff
@@ -1,65 +1,88 @@
-        raise ValueError("Input not from SQS")
-
-    # TODO: a large number of input collections will cause a timeout
-    # find a way to process each input message, deleting it from the queue
-    # any not processed before timeout will be retried on the next execution
     payloads = []
-    for record in [json.loads(r['body']) for r in event['Records']]:
-        payload = json.loads(record['Message'])
-        logger.debug('payload: %s', json.dumps(payload))
-        # expand payload_ids to full payloads
-        if 'payload_ids' in payload:
-            _payloads = ProcessPayloads.from_payload_ids(payload['payload_ids'])
-            if 'process_update' in payload:
-                logger.debug(
-                    "Process update: %s",
-                    json.dumps(payload['process_update']),
-                )
-                for c in _payloads:
-                    c['process'] = dict_merge(
-                        c['process'],
-                        payload['process_update'],
-                    )
-            ProcessPayloads(_payloads).process(replace=True)
-        elif payload.get('type', '') == 'Feature':
-            # If Item, create ProcessPayload and
-            # use default process for that collection
-            if payload['collection'] not in PROCESSES.keys():
-                raise ValueError(
-                    "Default process not provided for "
-                    f"collection {payload['collection']}",
-                )
-            payload_json = {
-                'type': 'FeatureCollection',
-                'features': [payload],
-                'process': PROCESSES[payload['collection']]
-            }
-            payloads.append(ProcessPayload(payload_json, update=True))
-        else:
+    failures = []
+    messages = {}
+    for message in utils.normalize_event(event):
+
+        try:
+            payload = utils.extract_record(message)
+        except Exception as e:
+            logger.exception('Failed to extract record: %s', json.dumps(message))
+            failures.append(message)
+
+        # if the payload has a URL in it then we'll fetch it from S3
+        try:
+            payload = utils.payload_from_s3(payload)
+        except NoUrlError:
+            pass
+
+        logger.debug('payload: %s', defer(json.dumps, payload))
+
+        try:
             payloads.append(ProcessPayload(payload, update=True))
+        except Exception:
+            logger.exception('Failed to convert to ProcessPayload: %s', json.dumps(payload))
+            failures.append(payload)
+
+        if is_sqs_message(message):
+            try:
+                messages[payload['id']].append(message)
+            except KeyError:
+                messages[payload['id']] = [message]
 
+    processed_ids = set()
     if len(payloads) > 0:
-        _payloads = ProcessPayloads(payloads)
-        _payloads.process()
+        processed = ProcessPayloads(payloads).process()
+        logger.error(processed)
```

Comment:
is error the right level for these? I think it would also help here if there explanatory text in the log message for what was being logged, rather than just dumping the objects into the log

Reply:
No, this was me being fast and sloppy with debug logging. Thanks for noticing.